Lib/fontTools/designspaceLib/split.py

```diff
@@ -352,9 +352,10 @@ def maybeExpandDesignLocation(object):
 def _conditionSetFrom(conditionSet: List[Dict[str, Any]]) -> ConditionSet:
     c: Dict[str, Range] = {}
     for condition in conditionSet:
+        minimum, maximum = condition["minimum"], condition["maximum"]
```

Comment:
should this allow for the "maximum" or "minimum" keys to actually be missing from the `condition` dict? Or are they guaranteed to be there, sometimes with a `None` value? I'm actually not sure who creates these `condition` dicts. But it's probably safe to do it nonetheless, as that seemed to be the original intent of this code.

```suggestion
        minimum, maximum = condition.get("minimum"), condition.get("maximum")
```

Reply:
Hi @anthrotype :)

These are the condition dicts from `DesignSpaceDocument.rules`. The keys are guaranteed to be there [when created through designspace parsing](https://github.com/fonttools/fonttools/blob/1306a71/Lib/fontTools/designspaceLib/__init__.py#L1838-L1850), and varLib [assumes that they are always there](https://github.com/fonttools/fonttools/blob/bb6fd8d22e7aa20d8ae6553c15f5686fbaf7d862/Lib/fontTools/varLib/__init__.py#L670-L677) too. This is what made me feel comfortable taking a similar approach.

With that said, other areas of the code, such as [rule evaluation](https://github.com/fonttools/fonttools/blob/1306a71/Lib/fontTools/designspaceLib/__init__.py#L395-L402) and [writing to XML](https://github.com/fonttools/fonttools/blob/1306a71/Lib/fontTools/designspaceLib/__init__.py#L1422-L1430), do perform this extra check.

I agree that there is no harm in including it, as it feels like a stronger guarantee of safety to me too, and bears no cost :+1: With that said, we should be mindful that other areas (e.g. varLib) might be more picky about the dicts they take.

Shall I add it in?

Reply:
yeah, let's use `dict.get` thanks

Reply:
Super! I have added the checks, and a few test cases to make sure this stays somewhat supported too.